frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -1184,7 +1126,38 @@ export default ({
         e?.dataTransfer.files?.length &&
           this.$refs.sendArea.fileAttachmentHandler(e?.dataTransfer.files, true)
       }
-    }
+    },
+    processSwitchQueue: debounce(async function () {
+      if (this.ephemeral.chatroomSwitchQueue.length === 0) return
+
+      // Take the most recent chatroom entry on the queue and discard everything else. This way we can speed up the process of switching chatrooms.
+      const targetChatroomId = this.ephemeral.chatroomSwitchQueue.pop()
```

Comment:
I wonder if maybe this couldn't be:

```js
const targetChatroomId = new String(this.ephemeral.chatroomSwitchQueue.pop())
```

This way, we can tell apart switching back to a chatroom from never having switched at all. At this point, this just aspirational.

Reply:
Personally I think it's OK because we have checks elsewhere to ensure that `processSwitchQueue` is only called when the channel switches